### PR TITLE
add return_tensor parameter for feature extraction

### LINKED ISSUE #10016: Feature-extraction pipeline to return Tensor
# 🚀 Feature request

Actually, to code of the feature-extraction pipeline 
`transformers.pipelines.feature-extraction.FeatureExtractionPipeline l.82` return a `super().__call__(*args, **kwargs).tolist()`

Which gives a list[float] (or list[list[float]] if list[str] in input)

I guess it's to be framework agnostic, but we can specify `framework='pt'` in the pipeline config so I was expecting a `torch.tensor`. 

Could we add some logic to return tensors ? 

# Motivation 
Features will be used as input of other models, so keeping them as tensors (even better on GPU) would be profitable. 

Thanks in advance for the reply, 

Have a great day.

### PR DESCRIPTION
**Pull request by ajsanjoaquin**
# What does this PR do?

Fixes #10016

## Before submitting
- [ x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [✔️](https://emojipedia.org/check-mark/#:~:text=See%20also%3A%20%E2%9C%85%20Check%20Mark,to%20Emoji%201.0%20in%202015.) Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [✔️ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [✔️ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x ] Did you write any new necessary tests?

Addresses [stale issue #10016](https://github.com/huggingface/transformers/issues/10016). Please review @LysandreJik and @Narsil. Thanks.

### CODE REVIEW

File: src/transformers/pipelines/feature_extraction.py [hunk 1]
```diff
@@ -39,6 +41,9 @@ class FeatureExtractionPipeline(Pipeline):
             Device ordinal for CPU/GPU supports. Setting this to -1 will leverage CPU, a positive will run the model on
             the associated CUDA device id.
     """
+    def __init__(self, return_tensors=False, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.return_tensors = return_tensors
```
Comment (Narsil) [inline src/transformers/pipelines/feature_extraction.py:46]: This needs to be moved in `_sanitize_parameters` to be consistent with all arguments of pipelines.

File: src/transformers/pipelines/feature_extraction.py [hunk 2]
```diff
@@ -61,6 +66,8 @@ def _forward(self, model_inputs):
 
     def postprocess(self, model_outputs):
```
Comment (Narsil) [inline src/transformers/pipelines/feature_extraction.py:67]: The default argument needs to be here. 

Check out https://huggingface.co/docs/transformers/v4.22.2/en/add_new_pipeline#how-to-create-a-custom-pipeline for more info on the code structure.

**Review by Narsil — COMMENTED** [review]
All up for it.

This PR needs to add some kind of test to make sure this argument works

Ideally in `tests/pipelines/test_pipelines_feature_extraction.py::run_pipeline_test` that way we make sure it works for all pipeline that are supposed to work.

File: src/transformers/pipelines/feature_extraction.py (hunk 1 above)
Comment (ajsanjoaquin) [reply src/transformers/pipelines/feature_extraction.py]: I have done everything you suggested. Test runs too.

**Comment by ajsanjoaquin** [comment]
Feedback has been addressed :)

File: src/transformers/pipelines/feature_extraction.py [hunk 3]
```diff
@@ -59,8 +63,10 @@ def _forward(self, model_inputs):
         model_outputs = self.model(**model_inputs)
         return model_outputs
 
-    def postprocess(self, model_outputs):
+    def postprocess(self, model_outputs, return_tensors=False):
         # [0] is the first available tensor, logits or last_hidden_state.
+        if self.return_tensors:
```
Comment (Narsil) [inline src/transformers/pipelines/feature_extraction.py:68]:
```suggestion
        if return_tensors:
```
No ?

**Comment by Narsil** [comment]
Not sure what's happening with CircleCI...

**Comment by ajsanjoaquin** [comment]
Last suggestion was addressed, but CircleCI Pipeline still seems broken...

**Comment by Narsil** [comment]
@ajsanjoaquin do you have circleCI set up on your account/branch ? I remember people sometimes having issues because of this.

Otherwise, could you try to rebase on `main`? 

Pinging @LysandreJik that might know more.

**Comment by LysandreJik** [comment]
The CircleCI pipeline seems to be working again!

**Comment by ajsanjoaquin** [comment]
@Narsil when running `make fixup`, I get the following error. Perhaps it's possible to replicate on your end?

-n was unexpected at this time.
make: *** [Makefile:10: modified_only_fixup] Error 255

**Comment by Narsil** [comment]
> -n was unexpected at this time.

What OS/Linux flavor you're running on ? Shell maybe ? 
Seems like `test -n` is not supported in your environment.

**Comment by ajsanjoaquin** [comment]
I'm in a temporary dev environment, so I'm using Windows 😅. I'll use WSL next time. So the formatting error was simply a result of missing newline before importing torch and tf?

**Comment by Narsil** [comment]
> I'm in a temporary dev environment, so I'm using Windows sweat_smile. I'll use WSL next time. So the formatting error was simply a result of missing newline before importing torch and tf?

We use `black` and `isort` to normalize as much as possible the code, so yes import order is automatically done for you.

**Comment by ajsanjoaquin** [comment]
@Narsil can this PR be merged soon when it passes all tests? I just resolved a conflict made by a different commit on the same file.

**Comment by Narsil** [comment]
@ajsanjoaquin 
It seems t[here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-) is an issue with your CircleCI permissions, the tests won't run.
Could you try refreshing your permissions as shown here?

**Comment by ajsanjoaquin** [comment]
@Narsil the `check_repository_consistency` test is failing because of a .md file unrelated to my PR. Do I have to address this in this PR?

`check_code_quality` is again failing and I can't use `make fixup`. I request you re-fix the latter. 

I hope this can be passed soon before another PR like #19382 modifies the exact code I was working on...

**Review by Narsil — APPROVED** [review]
LGTM.

I took the liberty of rebasing and fixing your PR. Is the result OK for you ?

@sgugger for final review.

**Review by sgugger — APPROVED** [review]
Thanks for adding this! Just one nit on the docstring.

**Comment by sgugger** [comment]
This PR was based on an old fork of the repo and as such, the pipeline tests were not run. They do not pass, so I will revert the commit. Could you open a new PR with fixed tests?

**Comment by Narsil** [comment]
I am confused, should we re-open a PR or start from here https://github.com/huggingface/transformers/pull/19679

(I tried re-opening a PR but it resulted in a no-op...)
